modules/plugin/csv/src/test/java/org/geotools/data/csv/TestCRSHandling.java

```diff
@@ -0,0 +1,105 @@
+
+    @BeforeClass
+    public static void createTemporaryLocations() throws IOException {
+        tmp = File.createTempFile("example", "");
+        boolean exists = tmp.exists();
+        if (exists) {
+            tmp.delete();
+        }
+        boolean created = tmp.mkdirs();
+        if (!created) {
+            System.exit(1);
+        }
+        statesfile = new File(tmp, "locations.csv");
+
+        URL resource = TestData.getResource(CSVWriteTest.class, "locations.csv");
+        Files.copy(resource.openStream(), statesfile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        Map<String, Object> params = new HashMap<>();
+        params.put(CSVDataStoreFactory.FILE_PARAM.key, statesfile.toString());
+        stateStore = DataStoreFinder.getDataStore(params);
+    }
+
+    @Test
+    public void testPrjFileRead()
+            throws FileNotFoundException, IOException, NoSuchAuthorityCodeException,
+                    FactoryException {
+        File f = TestData.file(this, "coastal2.csv");
+        Map<String, Object> params = new HashMap<>();
+        params.put(CSVDataStoreFactory.FILE_PARAM.key, f.toString());
+        params.put(CSVDataStoreFactory.STRATEGYP.key, CSVDataStoreFactory.WKT_STRATEGY);
+        params.put(CSVDataStoreFactory.WKTP.key, "WKT");
+
+        DataStore store = DataStoreFinder.getDataStore(params);
+        String name = store.getTypeNames()[0];
+        CoordinateReferenceSystem crs =
+                store.getFeatureSource(name).getSchema().getCoordinateReferenceSystem();
+        CoordinateReferenceSystem expected = CRS.decode("EPSG:27700");
+        assertEquals(
+                expected.getIdentifiers().iterator().next().getCode(),
+                crs.getIdentifiers().iterator().next().getCode());
+    }
+
+    @Test
+    public void testWritePrj() throws IOException {
+        File f = new File(tmp, "testcrs.csv");
+        Map<String, Object> params = new HashMap<>();
+        params.put(CSVDataStoreFactory.FILE_PARAM.key, f.toString());
+        params.put(CSVDataStoreFactory.STRATEGYP.key, CSVDataStoreFactory.WKT_STRATEGY);
+        params.put(CSVDataStoreFactory.WKTP.key, "WKT");
+        params.put(CSVDataStoreFactory.WRITEPRJ.key, "true");
+        DataStore store = DataStoreFinder.getDataStore(params);
+        store.createSchema(stateStore.getSchema(stateStore.getTypeNames()[0]));
+        SimpleFeatureSource source = store.getFeatureSource(store.getTypeNames()[0]);
+        if (!(source instanceof SimpleFeatureStore)) {
+            fail("can't create output file");
+        }
+        SimpleFeatureStore outstore = (SimpleFeatureStore) source;
+        outstore.addFeatures(
+                stateStore.getFeatureSource(stateStore.getTypeNames()[0]).getFeatures());
+        store.dispose();
+        String prjName = FilenameUtils.getBaseName(f.getName()) + ".prj";
+        File prj = new File(f.getParent(), prjName);
+        assertTrue(prj.exists());
```

Comment:
Suggestion, check that the contents can be read and have the expected value. Maybe by just round tripping, that, is, read back the file with a new store instance and checking that the CRS associated to the feature type is the expected one.

Reply:
Just as well I checked - it was wrong :-(